lib/licensing.js

```diff
@@ -130,10 +130,67 @@ const listSubComponents = (baseDir) => {
   return subComponents
 }
 
+const listNtpBackgrounds = (metadataFile) => {
+  const metadata = fs.readFileSync(metadataFile).toString()
+  let images = []
+  // Hack to import this TypeScript file in Node:
+  eval(metadata.replace('export const images: NewTab.Image[]', 'images'))
+  return images
+}
+
+const generateNtpBackgroundsLicense = (preamble, components) => {
+  let componentNotices = ''
+
+  for (let component of components) {
+    if (componentNotices.length != 0) {
+      componentNotices += '\n'
+    }
+
+    const fileName = component['source']
+    if (!fileName) {
```

Comment:
could we pull this logic out to a method? ex: 

```
const getValidatedComponentField = (component, fieldName) => {
    const fieldValue = component[fieldName]
    if (!fieldValue) {
       console.error('Missing' + fieldName + ' for background image ' + component['name'])
       process.exit(1)
    }
    return fieldValue
}
```

So it would condense down to:
const authorName = getValidatedComponentField(component, 'author')
const authorLink = getValidatedComponentField(component, 'link')

Reply:
I had some comments too that I removed, but we actually want to just move this to python/gn

Reply:
Good idea, I may as well since I have to rebase and repush to try and get CI to pass.